### ISSUE TITLE
Get rid of UNSUBSCRIBE_EXCEPTION

### ISSUE THREAD

**Issue by adamvy** [foam-framework/foam#528]
Use some programatic means of unsubscribing.

**Comment by kgrgreer** [comment]
Pass subscription or subscription destructor as extra argument to listeners so that they can easily unsubscribe without throwing an Exception.  Will avoid the use of non-error Exceptions and will be faster because try/catch blocks disable V8 JIT.

**Comment by shepheb** [comment]
Isn't it only `finally` that destroys optimizations? In any case, we should do this.

**Comment by kgrgreer** [comment]
No, try/catch is sufficient: http://www.html5rocks.com/en/tutorials/speed/v8/
